hat/assets/js/apps/Iaso/domains/workflows/components/changes/Modal.tsx

```diff
@@ -12,30 +13,33 @@ import {
     AddButton,
     LoadingSpinner,
 } from 'bluesquare-components';
-import ArrowRightAltIcon from '@material-ui/icons/ArrowRightAlt';
+import uniqWith from 'lodash/uniqWith';
+import isEqual from 'lodash/isEqual';
 
-import { Grid, Box, makeStyles } from '@material-ui/core';
-
-import InputComponent from '../../../../components/forms/InputComponent';
+import { Box, makeStyles } from '@material-ui/core';
 import { EditIconButton } from '../ModalButtons';
 import { MappingTable } from './MappingTable';
 import { Popper } from './InfoPopper';
 
-import { useGetForms } from '../../hooks/requests/useGetForms';
 import { useSaveWorkflowChange } from '../../hooks/requests/useSaveWorkflowChange';
 
 import MESSAGES from '../../messages';
 
 import { Change, Mapping, ReferenceForm } from '../../types';
 import { PossibleField } from '../../../forms/types/forms';
-import { useGetPossibleFields } from '../../../forms/hooks/useGetPossibleFields';
+import {
+    FormVersion,
+    useGetPossibleFieldsByFormVersion,
+} from '../../../forms/hooks/useGetPossibleFields';
+import { DropdownOptions } from '../../../../types/utils';
 
 type Props = {
     isOpen: boolean;
     closeDialog: () => void;
     change?: Change;
     versionId: string;
-    targetPossibleFields: PossibleField[];
+    targetPossibleFields?: PossibleField[];
+    targetPossibleFieldsByVersion?: FormVersion[];
```

Comment:
I wonder if you should not keep `targetPossibleFields` not nullable and pass a nullable version name rather than have the logic of what to choose inside the modal.